runtime/lua/vim/shared.lua

```diff
@@ -56,9 +49,13 @@ end
 ---
 ---@generic T: table
 ---@param orig T Table to copy
+---@param noref? boolean
+---When `false` a contained table is only copied once and all references point
+---to this single copy. When `true` every occurrence of a table results in a new
+---copy. This also means that a cyclic reference to `deepcopy()` can fail.
```

Comment:
Might be worth including when/why one might want to use `true` (i.e. the same text in the PR description, if a table contains unique/non-recursive values).